include/nil/crypto3/marshalling/zk/types/plonk/assignment_table.hpp

```diff
@@ -45,104 +45,193 @@ namespace nil {
+                            nil::marshalling::types::integral<TTypeBase, std::size_t>>>;
 
-                    std::vector<typename PlonkTable::field_type::value_type> table_values;
-                    for( std::size_t i = 0; i < PlonkTable::arithmetization_params::witness_columns; i++ ){
-                        for( std::size_t j = 0; j < assignments.rows_amount(); j++ ){
-                            table_values.push_back(assignments.witness(i)[j]);
+                    field_element_vector_type result;
+                    result.value().reserve(size * columns.size());
+                    for (std::size_t column_number = 0; column_number < columns.size(); column_number++) {
+                        std::size_t i = 0;
+                        for (; i < columns[column_number].size(); i++) {
+                            result.value().push_back(field_element_type(columns[column_number][i]));
                         }
-                    }
-                    for( std::size_t i = 0; i < PlonkTable::arithmetization_params::public_input_columns; i++ ){
-                        for( std::size_t j = 0; j < assignments.rows_amount(); j++ ){
-                            table_values.push_back(assignments.public_input(i)[j]);
+                        for (; i < size; i++) {
+                            result.value().push_back(field_element_type(padding));
                         }
                     }
-                    for( std::size_t i = 0; i < PlonkTable::arithmetization_params::constant_columns; i++ ){
-                        for( std::size_t j = 0; j < assignments.rows_amount(); j++ ){
-                            table_values.push_back(assignments.constant(i)[j]);
-                        }
+                    return result;
+                }
+
+                template<typename FieldValueType, typename Endianness>
+                std::vector<std::vector<FieldValueType>> make_field_element_columns_vector(
+                    const nil::marshalling::types::array_list<
+                        nil::marshalling::field_type<Endianness>,
+                        field_element<nil::marshalling::field_type<Endianness>, FieldValueType>,
+                        nil::marshalling::option::sequence_size_field_prefix<
+                            nil::marshalling::types::integral<nil::marshalling::field_type<Endianness>, std::size_t>>>
+                        &field_elem_vector,
+                    const std::size_t columns_amount,
+                    const std::size_t rows_amount) {
+
+                    std::vector<std::vector<FieldValueType>> result;
+                    result.resize(columns_amount);
+                    for (std::size_t i = 0; i < columns_amount; i++) {
+                        result[i].resize(rows_amount);
+                    }
+                    if (field_elem_vector.value().size() != columns_amount * rows_amount) {
+                        throw std::invalid_argument(
```

Comment:
Does this need to be a "throw" or an assert will work? In case the user has no way of passing some invalid argument and causing this and this can only happen because we call this function incorrectly, then consider BOOST_ASSERT_MSG(...).

Reply:
Changed to assert, I've used throw here because it seemed like this is the way errors are handled in marshalling code.